.evergreen/buildvariants-and-tasks.in.yml

```diff
@@ -91,6 +91,19 @@ const TEST_PACKAGED_APP_BUILD_VARIANTS = [
     run_on: 'macos-1100-gui',
     patchable: false,
     depends_on: 'package-macos-x64'
+  },
+  {
+    name: 'test-server-macos-14-arm',
```

Comment:
NIT: I think this could be an accidental replacement, could you do a drive-by and also rename the TEST_PACKAGED_APP_BUILD_VARIANTS to `test-packaged-app-XYZ`? `test-server-x` is meant for server tests and runs `test-server-x` tasks, these ones run `test-packaged-app`.